src/shared/Microsoft.Git.CredentialManager/JsonWebTokenExtensions.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+using System;
+using System.Security.Claims;
+using Microsoft.IdentityModel.JsonWebTokens;
+
+namespace Microsoft.Git.CredentialManager
+{
+    public static class JsonWebTokenExtensions
```

Comment:
Yes, we use it to trace the subject of the Azure access token when doing an new authentication.

```csharp
// Get an AAD access token for the Azure DevOps SPS
Context.Trace.WriteLine("Getting Azure AD access token...");
JsonWebToken accessToken = await _msAuth.GetAccessTokenAsync(
    authAuthority,
    AzureDevOpsConstants.AadClientId,
    AzureDevOpsConstants.AadRedirectUri,
    AzureDevOpsConstants.AadResourceId,
    remoteUri,
    null);

string atUser = accessToken.GetAzureUserName();
Context.Trace.WriteLineSecrets($"Acquired Azure access token. User='{atUser}' Token='{{0}}'", new object[] {accessToken.EncodedToken});
```

For the VS requested 'AT-only mode' work I also needed the user name to remember who was 'signed in'. Since that work is on hold now, we only use this in tracing. I think this is still worth it however since that extra information in the trace output can help diagnose multi-user problems.